packages/context/src/inject.ts

```diff
@@ -133,6 +133,11 @@ export function inject(
   if (injectionMetadata.bindingComparator && !resolve) {
     throw new Error('Binding comparator is only allowed with a binding filter');
   }
+  if (!bindingSelector && typeof resolve !== 'function') {
+    throw new Error(
```

Comment:
is it why we still throw Error here?
> I was planning to extend the message with such information, but it breaks a lot of existing test assertions

Reply:
This is fine as the error will be thrown when the class is loaded. There is no resolution context involved.